### PR TITLE
[Bug fix] LPS-143660 Rebase 12-31

### PR DESCRIPTION
**Pull request by dantewang**
Rebased. The conflict was just packageinfo and bnd files. @tinatian

### CODE REVIEW

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test

File: modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/JarUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@RunWith(Arquillian.class)
+public class JarUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testDownloadMySQLJarFromPortalProperties() throws Exception {
```
Comment (tinatian) [inline modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/JarUtilTest.java:55]: @dantewang Please either make this a unit test or a test for DataSourceFactoryImpl#testDatabaseClass where downloading mysql jar. I do not like the util test logic depend on property values.

Tina.

**Comment by tinatian** [comment]
@dantewang Please see my comment and resend.

**Comment by tinatian** [comment]
ci:close

**Pull request closed without merging**